logging/common/src/main/java/io/helidon/logging/common/BaseMdcProvider.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.logging.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import io.helidon.logging.common.spi.MdcProvider;
+
+/**
+ * Implements {@link io.helidon.logging.common.spi.MdcProvider} using an internal thread local data structure.
+ */
+public class BaseMdcProvider implements MdcProvider {
+
+    /*
+    Each instance of this class gets its own thread local map rather than usual practice of declaring thread local fields as
+    static. That's because each instance should represent a different provider type and there should be exactly one instance
+    for each provider type. This way each different provider type has its own thread local map in case, for some reason,
+    the maps for different providers need to be different.
+     */
+    private final ThreadLocal<Map<String, Supplier<String>>> mdcSettings = ThreadLocal.withInitial(HashMap::new);
```

Comment:
Thread local should be (by definition) static.

Reply:
TL;DR - This will no longer be an issue if my planned restructuring works as I think it will.

____
Thread local is not static by definition, but certainly in most cases is it static.

As the comments in the code explain, in this case I was looking to maintain a separate thread local map for each provider. With that in mind, it made sense to have the thread local map be an instance field. 

After thinking about this some more we really do not need separate thread local maps for each provider, so in the restructuring I have in mind `HelidonMdc` will maintain a single thread local map and it will be static there.

Reply:
No longer an issue with the rewrite of the PR.